esinstall/src/entrypoints.ts

```diff
@@ -0,0 +1,268 @@
+import fs from 'fs';
+import path from 'path';
+import validatePackageName from 'validate-npm-package-name';
+import {ExportField, ExportMapEntry, PackageManifestWithExports, PackageManifest} from './types';
+import {parsePackageImportSpecifier, resolveDependencyManifest} from './util';
+
+// Rarely, a package will ship a broken "browser" package.json entrypoint.
+// Ignore the "browser" entrypoint in those packages.
+const BROKEN_BROWSER_ENTRYPOINT = ['@sheerun/mutationobserver-shim'];
+
+function hasBrokenBrowserEntrypoint(packageName?) {
```

Comment:
also, helper functions that handle "undefined" arguments like this can be confusing. Can we instead only handle a valid string, and then only call this if `packageName` is defined?

```
function hasBrokenBrowserEntrypoint(packageName: string) 

if (packageName && hasBrokenBrowserEntrypoint(packageName)) {
  // do something special
```

```

Reply:
The reason I added this helper function was because the if statement was getting too complicated, but I get your point, I'll probably just remove it.

Reply:
I've removed it.